### DISCUSSION TITLE
"Back to Top" Button flickers while scrolling toward the top of page

### DISCUSSION THREAD

**Discussion by holmanb** [pradyunsg/furo#592] [Potential Issue] [answered]
Cloud-init's [docs](https://cloudinit.readthedocs.io/en/latest/topics/tutorials/lxd.html) recently [switched to furo](https://github.com/canonical/cloud-init/commit/c07453632d8f2c6f5885c437a331954a449b73cc). I just noticed while testing some changes on mobile (Android) that my average read/scroll speed apparently works out to be such that the button flickers while I'm scrolling. Not a huge deal, but pretty annoying.

I'm not sure if this is expected - on desktop Firefox this button only appears while scrolling upwards, which I'm not doing on mobile.

I'm not sure whether this is a weird mobile interaction, a bug, or if
it would be preferable to just debounce the "Back to Top" button so that it doesn't flicker. I doubt that this solution is preferred, but it would stop the flickering which would be "good enough" for me.

Has anyone else noticed this?

**Comment by pradyunsg** [comment]
What mobile browser are you using, what's the screen resolution and [what's the device pixel ratio](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio#result)?

I've seen this in the past with Safari on iOS, but not on Firefox for Android. I wouldn't be surprised if there's some weirdness around how the scroll position on the page is handled by mobile browsers; especially on hi-dpi screens.

**Reply by holmanb** [reply]
> What mobile browser are you using

Firefox

> what's the screen resolution

1080 x 2340

> what's the device pixel ratio

2.609

**Reply by pradyunsg** [reply]
> > what's the device pixel ratio
>
> 2.609

Wow. TIL this can be a default value!

**Comment by pradyunsg** [comment]
Well, in case someone else can spot something off about the logic involved here:

https://github.com/pradyunsg/furo/blob/da0674a849b525e753363781b38c4b10b808dde1/src/furo/assets/scripts/furo.js#L19-L30

**Comment by holmanb** [comment]
> I wouldn't be surprised if there's some weirdness around how the scroll position on the page is handled by mobile browsers; especially on hi-dpi screens.

Now that you mention it, I notice that text also shifts vertically by a pixel when this happens. This shift is much less noticeable than a button flashing in and out of existence, which I was much more focused on.

**Comment by pradyunsg** [comment]
I'm able to reproduce this on my phone (3.000 DPR) if I scroll _really_ slowly.

**Comment by holmanb** [comment]
I wonder if this could be debounced by with something like: "track the last N positions, and don't add/remove unless these positions are all increasing/decreasing", using heuristics to find N.

**Comment by pradyunsg** [comment, marked as answer]
Closing in favor of https://github.com/pradyunsg/furo/issues/681, which is the tracking issue for fixing this. :)
